src/common/tracer.h

```diff
@@ -79,26 +80,16 @@ struct span_stub {
   bool IsRecording() { return false; }
 };
 
-class jspan {
-  span_stub span;
- public:
-  span_stub& operator*() { return span; }
-  const span_stub& operator*() const { return span; }
-
-  span_stub* operator->() { return &span; }
-  const span_stub* operator->() const { return &span; }
-
-  operator bool() const { return false; }
-};
+using jspan_ptr = std::shared_ptr<jspan>; 
```

Comment:
these wrappers avoided shared_ptr before. can we just rename jspan->jspan_ptr and span_stub->jspan?

Reply:
when `WITH_JAEGER=ON` but tracing is disabled, isn't the shared_ptr always initialized with a 'noop span' rather than empty?

if this stub has different semantics, i worry that code that is tested and working when WITH_JAEGER=ON will crash with null pointer exceptions when built with OFF